source/extensions/http/stateful_session/header/header.h

```diff
@@ -54,9 +54,9 @@ class HeaderBasedSessionStateFactory : public Envoy::Http::SessionStateFactory {
     return !address.empty() ? absl::make_optional(std::move(address)) : absl::nullopt;
   }
 
-  std::string getHeaderName() const { return name_; }
+  Envoy::Http::LowerCaseString getHeaderName() const { return name_; }
```

Comment:
`const Envoy::Http::LowerCaseString&` to avoid unnecessary copy here.

Reply:
const won't allow that. Reaching out to discuss offline.

Reply:
`const Envoy::Http::LowerCaseString& getHeaderName() const { return name_; }`

Reply:
done